src/utils.coffee

```diff
@@ -4,13 +4,12 @@ utils =
   base64:
     encode: (data, to_encoding) ->
       return data if to_encoding is 'base64'
-      data = new Buffer data, to_encoding or 'utf8' unless Buffer.isBuffer data
+      data = new Buffer data.toString(), to_encoding or 'utf8' unless Buffer.isBuffer data
       data.toString 'base64'
     decode: (data, from_encoding) ->
       return data if from_encoding is 'base64'
-      data = (new Buffer data, 'base64')
-      return data unless from_encoding
-      data.toString from_encoding
+      data = (new Buffer data, 'base64') unless Buffer.isBuffer data
+      return data.toString from_encoding or 'utf8'
```

Comment:
yes. I just wanted to be consistent with the encode part, but sure, it can be ignored. What about the fallback 'utf8' if nothing else available?